snapshot/src/main/java/io/camunda/zeebe/snapshots/TransientSnapshot.java

```diff
@@ -24,4 +24,6 @@ public interface TransientSnapshot extends PersistableSnapshot {
    * @return true on success, false otherwise
    */
   ActorFuture<Void> take(Consumer<Path> takeSnapshot);
+
+  TransientSnapshot withLastFollowupEventPosition(long followupEventPosition);
```

Comment:
:wrench: please add some docs to clarify usage. It can be a bit confusing, especially when we set it to `lastWrittenPosition` in `AsyncSnapshotDirector`.